### ISSUE TITLE
`ngen.cal` cannot read `t-route` `netcdf`, `pickle`, `parquet`, and 3 `csv` variants

### ISSUE THREAD

**Issue by aaraney** [NOAA-OWP/ngen-cal#153]
Both the format and default naming scheme of `t-route`'s flow velocity and depth csv output format have changed. It seems these changes were introduced in https://github.com/NOAA-OWP/t-route/pull/762. Below is a truncated output from an experiment @ajkhattak was conducting:

Filename: `troute_output_201010010000.csv`
```csv
,,t0,time,flow,velocity,depth,nudge
6680,wb,2010-10-01 00:00:00,1:00:00,0.0,0.0,0.0,-9999.0
6693,wb,2010-10-01 00:00:00,1:00:00,0.011395932,0.14789358,0.016011778,-9999.0
6694,wb,2010-10-01 00:00:00,1:00:00,0.0011854391,0.046027187,0.0028443048,-9999.0
6695,wb,2010-10-01 00:00:00,1:00:00,0.0074746944,0.05957307,0.016491435,-9999.0
6696,wb,2010-10-01 00:00:00,1:00:00,0.00028689083,0.0100609,0.0011184122,-9999.0
6697,wb,2010-10-01 00:00:00,1:00:00,0.0010952102,0.017818118,0.00395199,-9999.0
6698,wb,2010-10-01 00:00:00,1:00:00,0.00067087985,0.0112075135,0.0036796732,-9999.0
6699,wb,2010-10-01 00:00:00,1:00:00,0.03966544,0.14632863,0.023701,-9999.0
6700,wb,2010-10-01 00:00:00,1:00:00,0.006763444,0.11990769,0.023327546,-9999.0
```

Related to #130

**Comment by aaraney** [comment]
Looks like they have slightly changed the format now https://github.com/NOAA-OWP/t-route/pull/818. Will need to account for both

**Comment by aaraney** [comment]
Looks like the new format will look like:

```csv
,,current_time,flow,velocity,depth,nudge
```

**Comment by aaraney** [comment]
Relevant things:

Of course the output you get from `t-route` depends on the configuration you provide. It seems there are multiple ways for `t-route` to generate flow, velocity, and depth outputs and you could write multiple outputs using different formats and / or table schemas.

Oldest supported `ngen.cal` `t-route` csv output configuration:

```yaml
output_parameters:
  csv_output:
    csv_output_folder: output/
```
This would write a single `csv` file in `output/` with either the name:
- `output/flowveldepth_{{network_topology_parameters.supernetwork_parameters.title_string}}.csv`
 if `network_topology_parameters.supernetwork_parameters.title_string` is not `None`
- `output/flowveldepth_{{datetime.now().isoformat()}}.csv`
 if `network_topology_parameters.supernetwork_parameters.title_string` _is_ `None`
 e.g. `output/flowveldepth_2024-08-07T15:24:08.440266.csv`


Using `stream_output`:

```yaml
output_parameters:
  stream_output: 
      stream_output_directory: output/
      stream_output_time: 10000000 # [hr]
      stream_output_type: '.csv'
      stream_output_internal_frequency: 60
```

Running `t-route` this will generate a `csv` file every `stream_output_time` (setting it higher than the simulation hours will only write to a single file). The name of the `csv` could be:

- `{{stream_output_directory}}/troute_output_{{t0}}.csv` in `%Y%m%d%H%M` format. `t0` can be calculated from several of the `compute_parameters.restart_parameters` fields:

```yaml
compute_parameters:
    restart_parameters:
        start_datetime              : 2023-04-02_00:00
        lite_channel_restart_file   : #restart/RESTART.2020082600_DOMAIN1
        lite_waterbody_restart_file : #restart/waterbody_restart_202006011200
```

This configuration in combination with above, would write to: `output/troute_output_202304020000.csv` where `t0` is calculated from `start_datetime`.

**Comment by aaraney** [comment]
Similar to `csv_output`, you could instead output `parquet` doing (although, this is not supported by `ngen.cal`):

```yaml
  parquet_output:
    parquet_output_folder: output/
    prefix_ids: wb-
```

**Comment by aaraney** [comment]
#162 adds support for `netcdf`, `pickle`, `parquet`, and 3 `csv` `t-route` variants.

**Comment by aaraney** [comment]
closed by #162
